tests/test_registry.py

```diff
@@ -101,6 +101,28 @@ def test_registry_deregister(registry, owner):
     assert tx2 == registry.getContract(name2)
     
 
+def test_register_edgecases(registry, owner):
+    name1 = s2b32("TestCoin1")
+    name2 = s2b32("TestCoin2")
+    
+    tx1 = TestCoin.deploy({'from': owner})
+    tx2 = TestCoin.deploy({'from': owner})
+
+    with brownie.reverts("ERROR:REC-011:RELEASE_UNKNOWN"):
+        registry.registerInRelease(s2b32("unknown release"), name1, tx1, {'from': owner})
+
+    with brownie.reverts("ERROR:REC-012:CONTRACT_NAME_EMPTY"):
+        registry.register("", tx1, {'from': owner})
+
+    registry.register(name1, tx1, {'from': owner})
+    with brownie.reverts("ERROR:REC-013:CONTRACT_NAME_EXISTS"):
+        registry.register(name1, tx2, {'from': owner})
+    
+    with brownie.reverts("ERROR:REC-014:CONTRACT_ADDRESS_ZERO"):
+        registry.register(name2, ZERO_ADDRESS, {'from': owner})
+

```

Comment:
also add deregister of a non-existing module as edge case

Reply:
thats already there in another test (`test_registry_deregister`) https://github.com/etherisc/gif-contracts/blob/7843a6b3f9f9e0d4c8bb2b1a8f819f4aad8ad9fb/tests/test_registry.py#L91